hw/dv/sv/test_vectors/test_vectors_pkg.sv

```
@@ -334,6 +347,24 @@ package test_vectors_pkg;
           "KeyLen": begin
             vector.key_length_word = entry_data.atoi() / 32;
           end
+          "Klen": begin
+            // rounding up key length to nearest predefined key length
+            if (entry_data.atoi() <= 16) begin
```

It might also make sense to check that the string is "0" if `entry_data.atoi()` returns 0 (raising an error otherwise). Otherwise, I think you'll silently ignore things like "cabbage".

---

Hmm, I'm not sure this is doing quite the right thing. How do I get to the trailing else in the `if` block now, representing a key bigger than 1024 bits?

---

I'm not sure I follow, in the last pushed version of the code, if `key_length_byte > 128`, I would end up in the last trailing else block where `vector.key_length_word = 0;` - why do I not get to it?

---

My question was a bit silly, sorry! Embarrassingly, I'd forgotten that there are numbers greater than 128 :-) Your code looks perfectly sensible to me!